contracts/ConsumerHost.sol

```diff
@@ -173,14 +173,18 @@ contract ConsumerHost is Initializable, OwnableUpgradeable, IConsumer, ERC165 {
      * @notice Deposit amount to hosting, need consumer approve firstly
      * @param amount the amount
      */
-    function deposit(uint256 amount) external {
+    function deposit(uint256 amount, bool approve) external {
         // transfer the balance to contract
         IERC20 sqt = IERC20(SQT);
         sqt.safeTransferFrom(msg.sender, address(this), amount);
         sqt.approve(channel, amount);
 
         Consumer storage consumer = consumers[msg.sender];
         consumer.balance += amount;
+        if (approve) {
```

Comment:
Why not include the `approve` function call in `deposit` directly, so that the caller doesn't have to concern whether have approved

Reply:
That would force approve every time, and I'd prefer to make approve optional to retain flexibility. If we want to need approved before use it, we can do it on the consumer host page.